src/main/kotlin/org/wycliffeassociates/sourceaudio/upload/RequestHandler.kt

```diff
@@ -0,0 +1,76 @@
+package org.wycliffeassociates.sourceaudio.upload
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.ktor.http.content.PartData
+import io.ktor.http.content.streamProvider
+import org.wycliffeassociates.sourceaudio.upload.model.FileUploadModel
+import java.io.File
+import java.lang.IllegalArgumentException
+
+class RequestHandler {
+
+    fun handleFormUpload(parts: List<PartData>): String {
+        val paramsMap: MutableMap<String, String> = mutableMapOf()
+        var file = File("./src/main/resources/uploads/")
+
+        parts.forEach { part ->
+            when (part) {
+                is PartData.FormItem -> {
+                    paramsMap[part.name!!] = part.value
+                }
+                is PartData.FileItem -> {
+                    file = file.resolve(part.originalFileName!!)
+                    part.streamProvider().use { input ->
+                        file.outputStream().buffered().use {
+                            input.copyTo(it)
+                        }
+                    }
+                }
+            }
+            part.dispose()
+        }
+
+        return processUploadModel(file, paramsMap)
+    }
+
+    private fun processUploadModel(file: File, params: Map<String, String>): String {
+        var response = mutableMapOf<String, Any>()
+        try {
```

Comment:
You have a large try body because the FileUploadModel validates upon initialization. Maybe we should change this? It's quite inconvenient to have to catch errors upon instantiating a data class. Let me know your thoughts on that. If we didn't validate upon instantiation, it would also make it easier to break up the rest of this function into more, smaller functions.